### DISCUSSION TITLE
WCAG 2.3.1 Three Flashes or Below Threshold

### DISCUSSION THREAD

**Discussion by evansque** [w3c/wcag#2286] [Q&A] [answered]
I am working on a flash detection tool based on WCAG SC 2.3.1
from [WCAG 2.3.1 SC](https://www.w3.org/WAI/WCAG21/Understanding/three-flashes-or-below-threshold.html)

> A general flash is defined as a pair of opposing changes in [relative luminance](https://www.w3.org/WAI/WCAG21/Understanding/three-flashes-or-below-threshold.html#dfn-relative-luminance) of 10% or more of the maximum relative luminance where the relative luminance of the darker image is below 0.80; and where "a pair of opposing changes" is an increase followed by a decrease, or a decrease followed by an increase

Do the increase/decrease values have to be sequential in a one second period?

Here is an example:
In a 25f/s video (f1, f2, f3, f4, f5, f6 ... f25). does the following changes in luminance values count as a flash?
f1-f2 4% increase
f2-f3 4% increase
f3-f4 4% increase
f4-f5 4% decrease

or does it have to be 10% or more from one frame to another?

**Comment by patrickhlauke** [comment]
the flash does not have to take place from one frame to another, no. you look at how long the change of at least 10% and then back took, and based on that duration work out the frequency, and then check that the frequency falls below 3Hz / 3 times a second.

so in your case (extending the decrease to go all the way back to the initial value)

> f1-f2 4% increase (from original value of 0%)
f2-f3 4% increase
f3-f4 4% increase (this is the point where it's 10% or more from original value)
f4-f5 4% decrease
f5-f6 4% decrease
f6-f7 4% decrease (back to original value of 0%)

we can say that the "pair of opposing changes" (i.e. the increase of 10% or more, and then the decrease back to original value) takes a total of 7 frames. that one single flash takes 0.28 seconds to complete (7 frames / 25 frames per second). if this is a continuous increase/decrease/increase/decrease (i.e. it keeps looping), then the flashing frequency here is 25 frames per second / 7 frames per flash = 3.57 flashes per second, which is faster than the limit of 3 flashes per second of the SC

**Comment by patrickhlauke** [comment, marked as answer]
if your question was more about "is this a flash" (i.e. increasing it by 4% three times, and decreasing it by just 4%), then no that exact sequence in itself is not a flash. a flash is a change of 10% or more one way, and then a change of 10% back the other way. the only "pair of opposing changes" in your sequence is an increase of 12% from f1-f4, and then a decrease of 4% from f4-f5. so this in itself is not a "pair of opposing changes" each of which are 10% or above.

**Reply by awkawk** [reply]
@patrickhlauke I'm not so sure. If there is a smooth transition from 0% to 10% and back in a nice smooth sine wave, and the cycle repeats in <3 seconds, it doesn't matter that each incremental step toward the top or bottom isn't 10%. The user perceives a flash, right?

**Reply by patrickhlauke** [reply]
@awkawk if it goes 0%-10%. then back down 10%-0%, sure that counts as a flash. but the OP was asking specifically about an increase of 12%, followed by a decrease from 12% down to 8% (and did not say anything about repeating/cycling). so that specifically I was saying does not count as a flash that violates the general flash definition (as it doesn't decrease strongly enough - it's more like an eased increase).

of course what you describe (smooth sine wave increase 0%-10%, then back down 10%-0%, then cycling) is perceived as a flash (based on the general flash definition as well), and yes it doesn't matter if it happens in a single frame, or many frames. as I said in my answer above  https://github.com/w3c/wcag/discussions/2286#discussioncomment-2456847

**Comment by Myndex** [comment]
Hi @evansque 

**It is not per frame, it is number of cycles per second from valley to peak.**

While the existing SC for 2.3.1 indicates a 10% change, and nothing faster than 3 per second, it is important to point out that these guidelines are based on research done mostly in the 1990s, and done using CRT televisions (typically 21") and in a living room setting.

Those CRTs had a maximum luminance of 80 to 100 cd/m²... today, mobile devices can be as bright as 1200 cd/m². and HDR screens can do up to 10,000 cd/m².

Today, the environment is completely different! Unfortunately we do not have much in the way of modern research using newer technology screens in the new and different ways that mobile devices are used.

I can tell you that there is some discussion on changing the rate guidelines to no more than 2 hz. 

And while 10% luminance change is fairly small, on a device that can output 1200 cd/m², 10% is 120 cd/m² which is more than 100% of an old 1990's era CRT TV.

### UPSHOT
Until there is more defined guidance from new research, IMO the safer approach is to simply consider the flashing rate, and the visual angle. Considering the use of mobile devices, an area larger than about 50px square and flashing faster than 3 Hz should be avoided, regardless of the peak to valley "percentage" change.

**Reply by patrickhlauke** [reply]
> **It is not per frame, it is number of cycles per second from valley to peak.**

just a tiny note here: not from valley to peak, but from peak to peak/valley to valley (a full waveform cycle) - the definition of flash is the *pair* of opposing changes https://www.w3.org/TR/WCAG21/#dfn-flashes
